[Map] Exception when moving point

On clicking green checkmark:

```
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug: io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.IllegalStateException: Cannot invoke setValue on a background thread
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.subscribers.LambdaSubscriber.onError(LambdaSubscriber.java:79)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.subscribers.LambdaSubscriber.onNext(LambdaSubscriber.java:69)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at com.uber.autodispose.HalfSerializer.onNext(HalfSerializer.java:55)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at com.uber.autodispose.AutoDisposingSubscriberImpl.onNext(AutoDisposingSubscriberImpl.java:118)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.FlowableOnAssembly$OnAssemblySubscriber.onNext(FlowableOnAssembly.java:61)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.mixed.FlowableSwitchMapSingle$SwitchMapSingleSubscriber.drain(FlowableSwitchMapSingle.java:243)
2021-06-22 18:56:56.631 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.mixed.FlowableSwitchMapSingle$SwitchMapSingleSubscriber$SwitchMapSingleObserver.onSuccess(FlowableSwitchMapSingle.java:277)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.SingleOnAssembly$OnAssemblySingleObserver.onSuccess(SingleOnAssembly.java:68)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.single.SingleOnErrorReturn$OnErrorReturn.onSuccess(SingleOnErrorReturn.java:82)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.SingleOnAssembly$OnAssemblySingleObserver.onSuccess(SingleOnAssembly.java:68)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableToSingle$ToSingle.onComplete(CompletableToSingle.java:68)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletablePeek$CompletableObserverImplementation.onComplete(CompletablePeek.java:115)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableFromRunnable.subscribeActual(CompletableFromRunnable.java:47)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.Completable.subscribe(Completable.java:2309)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly.subscribeActual(CompletableOnAssembly.java:39)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.Completable.subscribe(Completable.java:2309)
2021-06-22 18:56:56.632 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$NextObserver.onComplete(CompletableAndThenCompletable.java:99)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.onComplete(CompletableSubscribeOn.java:79)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableFromCallable.subscribeActual(CompletableFromCallable.java:47)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.Completable.subscribe(Completable.java:2309)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly.subscribeActual(CompletableOnAssembly.java:39)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.Completable.subscribe(Completable.java:2309)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
2021-06-22 18:56:56.633 6537-6890/com.google.android.gnd E/RxDebug:     at java.lang.Thread.run(Thread.java:764)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug: Caused by: java.lang.IllegalStateException: Cannot invoke setValue on a background thread
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at androidx.lifecycle.LiveData.assertMainThread(LiveData.java:462)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at androidx.lifecycle.LiveData.setValue(LiveData.java:304)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at com.google.android.gnd.ui.home.mapcontainer.MapContainerViewModel.setViewMode(MapContainerViewModel.java:355)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at com.google.android.gnd.ui.home.mapcontainer.MapContainerFragment.setDefaultMode(MapContainerFragment.java:316)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at com.google.android.gnd.ui.home.HomeScreenFragment.onFeatureUpdated(HomeScreenFragment.java:168)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at com.google.android.gnd.ui.home.HomeScreenFragment.lambda$nbskEKgDNMz4eC0cJ6Mc_GIE9ic(Unknown Source:0)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at com.google.android.gnd.ui.home.-$$Lambda$HomeScreenFragment$nbskEKgDNMz4eC0cJ6Mc_GIE9ic.accept(Unknown Source:4)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug:     at io.reactivex.internal.subscribers.LambdaSubscriber.onNext(LambdaSubscriber.java:65)
2021-06-22 18:56:56.634 6537-6890/com.google.android.gnd E/RxDebug: 	... 37 more
2021-06-22 18:56:57.144 1746-1768/system_process E/memtrack: Couldn't load memtrack module
2021-06-22 18:57:02.328 1746-1774/system_process E/BatteryStatsService: modem info is invalid: ModemActivityInfo{ mTimestamp=0 mSleepTimeMs=0 mIdleTimeMs=0 mTxTimeMs[]=[0, 0, 0, 0, 0] mRxTimeMs=0 mEnergyUsed=0}
```

--- a/gnd/src/main/java/com/google/android/gnd/persistence/local/LocalValueStore.java
+++ b/gnd/src/main/java/com/google/android/gnd/persistence/local/LocalValueStore.java
@@ -93,7 +93,11 @@ public class LocalValueStore {
 
   public Optional<CameraPosition> getLastCameraPosition(String projectId) {
     try {
-      String[] values = preferences.getString(LAST_VIEWPORT_PREFIX + projectId, "").split(",");
+      String value = preferences.getString(LAST_VIEWPORT_PREFIX + projectId, "");
+      if (value.isEmpty()) {
+        return Optional.empty();
+      }
+      String[] values = value.split(",");
       return Optional.of(
           new CameraPosition(
               Point.newBuilder()

--- a/gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureMutationEntity.java
+++ b/gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureMutationEntity.java
@@ -87,7 +87,7 @@ public abstract class FeatureMutationEntity extends MutationEntity {
         .setProjectId(getProjectId())
         .setFeatureId(getFeatureId())
         .setLayerId(getLayerId())
-        .setNewLocation(Optional.ofNullable(getNewLocation().toPoint()))
+        .setNewLocation(Optional.ofNullable(getNewLocation()).map(Coordinates::toPoint))
         .setType(getType().toMutationType())
         .setRetryCount(getRetryCount())
         .setLastError(getLastError())

--- a/gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java
+++ b/gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java
@@ -319,8 +319,8 @@ public class MapContainerViewModel extends AbstractViewModel {
   }
 
   public void setViewMode(Mode viewMode) {
-    mapControlsVisibility.setValue(viewMode == Mode.DEFAULT ? VISIBLE : GONE);
-    moveFeaturesVisibility.setValue(viewMode == Mode.REPOSITION ? VISIBLE : GONE);
+    mapControlsVisibility.postValue(viewMode == Mode.DEFAULT ? VISIBLE : GONE);
+    moveFeaturesVisibility.postValue(viewMode == Mode.REPOSITION ? VISIBLE : GONE);
   }
 
   public LiveData<Integer> getMapControlsVisibility() {

--- a/gnd/src/main/res/values-pt/strings.xml
+++ b/gnd/src/main/res/values-pt/strings.xml
@@ -158,7 +158,7 @@ Tem certeza que deseja apagar o local e todos os dados associados? Isso não pod
   <string name="offline_base_map_viewer_navgraph_label">Visualizador offline de base de mapas</string>
   <string name="offline_base_map_viewer_title">Visualizador offline de Títulos de Base de Mapas</string>
   <string name="offline_base_map_viewer_remove_button">Remover</string>
-  <string name="offline_base_map_viewer_storage">Essa base de mapa ocupará %. Lf MB do espaço no seu dispositivo.</string>
+  <string name="offline_base_map_viewer_storage">Essa base de mapa ocupará %.1f MB do espaço no seu dispositivo.</string>
   <string name="unnamed_area">Area sem nome</string>
   <string name="offline_base_map_download_started">Download iniciado</string>
   <string name="offline_base_map_download_failed">Erro no download da Base de  mapas</string>

